### PR TITLE
feat(call-home): added ci for call-home -transmitter

### PR DESCRIPTION
**Pull request by datacore-vkumar**
Added Ci for call-home-transmitter components

Note- It is dependent on call-home transmitter PR #20  
Signed-off-by: Vivek Kumar <vivek.kumar@datacore.com>

### CODE REVIEW

File: nix/pkgs/images/default.nix [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
   image_suffix = { "release" = ""; "debug" = "-debug"; "coverage" = "-coverage"; };
-  build-extensions-image = { pname, buildType, package, config ? { } }:
+  build-extensions-image = { pname, buildType, package, extraCommands ? '''', contents ? [ ], config ? { } }:
     dockerTools.buildImage {
       tag = extensions.version;
       created = "now";
       name = "mayadata/mayastor-${pname}${image_suffix.${buildType}}";
-      contents = [ package ];
+      contents = [ package ] ++ contents;
+      extraCommands = extraCommands;
```
Comment (tiagolobocastro) [inline nix/pkgs/images/default.nix:14]:
```suggestion
      dockerTools.buildImage {
        inherit extraCommands;
        
        tag = extensions.version;
        created = "now";
        name = "mayadata/mayastor-${pname}${image_suffix.${buildType}}";
        contents = [ package ] ++ contents;
```

File: nix/pkgs/extensions/default.nix [hunk 2]
```diff
@@ -44,6 +44,19 @@ let
         pname = "upgrade-operator";
       };
     };
+    call-home = rec {
```
Comment (tiagolobocastro) [inline nix/pkgs/extensions/default.nix:47]: should `call-home` be part of `exporter`?

Comment (datacore-vkumar) [reply nix/pkgs/extensions/default.nix:47]: In my opinion, No as call-home has a separate directory and it is not exposing any data through any HTTP endpoint instead it sends the encrypted JSON to a swarm server every 24 hours. So this data can not be read by Prometheus or in any other way by the user.

Comment (tiagolobocastro) [reply nix/pkgs/extensions/default.nix:47]: That's how it currently is (directory structure) but I'm questioning if that's how we really want it to be. 
I thought `exporter` was just a generic name, it's exporting things, and 1 example is `metrics`. As per the nomenclature in the wiki, this is our naming convention: `<classification>-<specific_name>`.
I could see call home here as `exporter.call-home`?

Comment (datacore-vkumar) [reply nix/pkgs/extensions/default.nix:47]: Yes, I got your point but this name was decided during the earlier discussions and it is also there in the system impact statement that's why I used this name.

**Review by tiagolobocastro — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Comment by datacore-vkumar** [comment]
bors try

**Comment by datacore-vkumar** [comment]
bors merge

**Pull request merged**